src/lib/ext-activitylog.js

```diff
@@ -194,6 +233,34 @@ class Controller {
     }
   }
 
+  async loadLogs(file) {
+    try {
+      const logStr = await load.loadLogAsText(file);
+      const logs = JSON.parse(logStr);
```

Comment:
Nit, we do have a load ESM module, I think that JSON.parse should be a responsibility of that module instead of being a responsibility of the Controller class.

Reply:
done :-)